main.go

```diff
@@ -461,11 +553,85 @@ func (svr *bridge) handleCall(w http.ResponseWriter, r *http.Request) {
 	http.Error(w, strings.Join(text, "\n"), respCode)
 }
 
+func parseMsgTemplates(tmplPath string, token string) (*ut.Template, error) {
+	var tmpl *ut.Template
+	var dirs []string
+	var tmplNames []string
+
+	err := filepath.Walk(tmplPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("File or Folder discovery issue: %s", err)
+		}
+		if !info.IsDir() {
+			filename := info.Name()
+			dupFileNames := contains(tmplNames, filename)
+			if dupFileNames {
+				return fmt.Errorf("Repeated user-defined template file names are not allowed: %s", filename)
+			}
+			tmplNames = append(tmplNames, filename)
+		} else {
+			dirs = append(dirs, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return tmpl, fmt.Errorf("Error: A user-defined template discovery contains errors: %s\n", err)
+	}
+
+	fileExt := []string{"html", "gohtml", "gotmpl", "tmpl"}
+	for _, p := range fileExt {
+		matchedTmpls, err := ut.ParseGlob(tmplPath + "/*." + p)
+		if err == nil {
+			tmpl = ut.Must(matchedTmpls, err)
```

Comment:
Yes, the two different `must` lines are required when parsing through root/subdirectories.

Reply:
OK, thanks. I asked because [the source for Must](https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/text/template/helper.go;l=24-29) just panics in case the err is not `nil` (which we guard against the line immediately before it). So, I think this is just a duplicate check.

I didn't elaborate well on why it was a concern. We should avoid panics during HTTP request servicing if possible. I can't remember if the panic gets `recover`ed by the servemux and just turns into an HTTP 500 to the user or if it actually crashes the whole process. I do recall that the failure never makes it to the log stream unless one writes some recovery middleware to catch it. So, in those scenarios (particularly for function calls), it's usually better to add error handling around the function call. In this case, we already have that since the second return of the function is an `error` and we're paying attention to it in the caller.

Reply:
As you read through that section, I think you are assuming the first `ParseGlob`/`Must` would parse all templates through the entire file/folder structure. When I started working with templates, I assumed the same thing.

The first, `ParseGlob`/`Must` checks the root folder for templates, and the second, `ParseGlob`/`Must`, checks any nested folders within the root for templates. Without the second, no templates in the nested folders will get pulled in.

We should be solid with no handling during the HTTP request since we moved the parsing of the user's templates during startup, so any errors would get triggered before the service started firing. If errors occur during the user template parsing, everything will revert to default before starting up HTTP processing.